src/main/java/com/commercetools/sync/producttypes/utils/ProductTypeUpdateAttributeDefinitionActionUtils.java

```diff
@@ -158,15 +159,44 @@
                 final AttributeDefinitionDraft matchingNewAttributeDefinitionDraft =
                     newAttributeDefinitionDraftsNameMap.get(oldAttributeDefinitionName);
                 return ofNullable(matchingNewAttributeDefinitionDraft)
-                    .map(attributeDefinitionDraft ->
-                        buildActions(oldAttributeDefinition, attributeDefinitionDraft)
-                    )
+                    .map(attributeDefinitionDraft -> {
+                        // attribute type is required so if null we let commercetools to throw exception
+                        if (attributeDefinitionDraft.getAttributeType() != null) {
+                            if (haveSameAttributeType(oldAttributeDefinition, attributeDefinitionDraft)) {
+                                return buildActions(oldAttributeDefinition, attributeDefinitionDraft);
+                            } else {
+                                return Arrays.asList(
+                                    RemoveAttributeDefinition.of(oldAttributeDefinitionName),
+                                    AddAttributeDefinition
+                                        .of(AttributeDefinitionCustomBuilder.of(attributeDefinitionDraft))
+                                );
+                            }
+                        } else {
+                            return new ArrayList<UpdateAction<ProductType>>();
+                        }
+
+                    })
```

Comment:
general questions about this method: Why does it take already a map `newAttributeDefinitionDraftsNameMap ` as a parameter and not a list? can't the map be already created inside of it   as it's only needed by it?

Reply:
6f72b6200fd20fa1017916e7be71e9880eb3fc8b